arches_lingo/src/arches_lingo/App.vue

```diff
@@ -22,32 +27,36 @@ provide(userKey, { user, setUser });
 
 const router = useRouter();
 const route = useRoute();
+const toast = useToast();
+const { $gettext } = useGettext();
 
-const isAuthenticated = computed(() => {
-    return user.value && user.value.username !== "anonymous";
-});
+router.beforeEach(async (to, _from, next) => {
+    try {
+        const userData = await fetchUser();
+        setUser(userData);
 
-router.beforeEach(async (to) => {
-    if (
-        !isAuthenticated.value &&
-        // Avoid an infinite redirect
-        to.name !== routeNames.login
-    ) {
-        return { name: routeNames.login, params: { next: to.name as string } };
+        const requiresAuth = to.matched.some(
+            (record) => record.meta.requiresAuth,
+        );
+        if (requiresAuth && userData.username === "anonymous") {
+            throw new Error();
+        } else {
+            next();
+        }
+    } catch (error) {
+        toast.add({
+            severity: ERROR,
+            life: DEFAULT_ERROR_TOAST_LIFE,
+            summary: $gettext("Login required."),
+            detail: error instanceof Error ? error.message : undefined,
+        });
```

Comment:
I like this in principle, but it's too much when just navigating to the home page and getting redirected to login.

Reply:
cool cool, removed the `requiresAuthentication` req for the home route 👍